uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/annual-entitlement-change-log.component.js

```diff
@@ -0,0 +1,319 @@
+     * @return {Promise}
+     */
+    function loadAbsenceTypes (calculationUnits) {
+      return AbsenceType.all().then(function (absenceTypes) {
+        return absenceTypes.map(function (absenceType) {
+          var unit = calculationUnits[absenceType.calculation_unit];
+
+          return _.extend({
+            'calculation_unit.name': unit.name,
+            'calculation_unit.label': unit.label
+          }, absenceType);
+        });
+      })
+      .then(function (absenceTypes) {
+        vm.absenceTypes = absenceTypes;
+      });
+    }
+
+    /**
+     * Returns a map of absence type calculation units indexed by their
+     * values.
+     *
+     * @return {Promise}
+     */
+    function loadCalculationUnits () {
+      return OptionGroup.valuesOf('hrleaveandabsences_absence_type_calculation_unit')
+      .then(function (calculationUnits) {
+        return _.indexBy(calculationUnits, 'value');
+      });
+    }
+
+    /**
+     * Loads and stores the Leave Entitlements change log for the contact and
+     * period provided.
+     *
+     * @return {Promise}
+     */
+    function loadChangeLog () {
+      return Entitlement.logs({
+        contact_id: vm.contactId,
+        period_id: vm.periodId
+      })
+      .then(function (changeLog) {
+        entitlementsChangeLog = changeLog;
+      });
+    }
+
+    /**
+     * Removes comments that remain the same from one row to the other.
+     * This reduces duplication of rows when displaying one row per comment.
+     */
+    function removeRepeatedComments () {
+      var currentLogRowPointer, nextLogRowPointer;
+      var logHasOneOrCeroRows = vm.changeLogRows.length <= 1;
+
+      // There is no chance of repetition when there are one or cero rows
+      if (logHasOneOrCeroRows) {
+        return;
+      }
+
+      currentLogRowPointer = vm.changeLogRows.length - 2;
+      nextLogRowPointer = vm.changeLogRows.length - 1;
+
+      while (currentLogRowPointer >= 0) {
+        var currentLogRow = vm.changeLogRows[currentLogRowPointer];
+        var nextLogRow = vm.changeLogRows[nextLogRowPointer];
+
+        currentLogRow.entitlements.forEach(function (currentLogRowEntitlement, i) {
+          var nextLogRowEntitlement = nextLogRow.entitlements[i];
+
+          if (currentLogRowEntitlement.comment === nextLogRowEntitlement.comment) {
+            delete currentLogRowEntitlement.comment;
+          }
+        });
+
+        currentLogRowPointer--;
+        nextLogRowPointer--;
+      }
+    }
+
+    /**
+     * When a row has multiple entitlements with comments this function can be
+     * used to create a new row for each of the comments. The new rows created
+     * will display the change between comments and highlight the comment's
+     * entitlement.
+     *
+     * @param {Number} rowIndex - The row index with multiple entitlement
+     * comments that will be split into different rows.
+     */
+    function splitEntitlementCommentsIntoMultipleRows (rowIndex) {
```

Comment:
This function looks a bit large. If it is not that difficult, could you break this function into separate ones?